### PR TITLE
Use Authorization header for BasicAuth

### PR DESCRIPTION
**Pull request by StykMartin**
Small change.
Use proper headers to accept usr/pass combination.

### CODE REVIEW

File: Server/bkr/server/authentication.py [hunk 1]
```diff
@@ -24,28 +24,25 @@ def login_password():
     The caller may act as a proxy on behalf of another user by passing the
     *proxy_user* key. This requires that the caller has 'proxy_auth'
     permission.
-    The request body must be a JSON object containing username and password.
+    The request must contain Authorization header that contains the word Basic
+    word followed by a space and a base64-encoded string username:password.
```
Comment (JohnVillalovos) [inline Server/bkr/server/authentication.py:28]: s/word followed/followed/

File: Server/bkr/server/authentication.py [hunk 2]
```diff
@@ -24,28 +24,25 @@ def login_password():
...
     proxy_user = payload.get("proxy_user")
+    username = request.authorization.username
+    password = request.authorization.password
 
     user = User.by_user_name(username)
-    if user is None:
-        raise Unauthorised401(u"Invalid username or password")
-    if not user.can_log_in():
+    if (user is None
+            or user.can_log_in()
```
Comment (JohnVillalovos) [inline Server/bkr/server/authentication.py:45]: Shouldn't there be a "not" here and the line below?

**Review by JohnVillalovos — CHANGES_REQUESTED** [review]

File: Server/bkr/server/authentication.py (hunk 2 above)
Comment (StykMartin) [reply Server/bkr/server/authentication.py:45]: :| Of course it should be like this. When users can log in, we refuse 😸 .
I will fix it.

**Review by JohnVillalovos — APPROVED** [review]

**Review by renanrodrigo — APPROVED** [review]
lgtm

**Pull request merged**